core/modules/mpls_pop.cc

```diff
@@ -0,0 +1,41 @@
+#include "mpls_pop.h"
+
+#include "../utils/ether.h"
+#include "../utils/mpls.h"
+
+
+void MPLSPop::ProcessBatch(bess::PacketBatch *batch) {
+  using bess::utils::be16_t;
+  using bess::utils::Ethernet;
+  using bess::utils::Mpls;
+
+  int cnt = batch->cnt();
+
+  for (int i = 0; i < cnt; i++) {
+    bess::Packet *pkt = batch->pkts()[i];
+
+    Ethernet *eth = pkt->head_data<Ethernet *>();
+
+    if (eth->ether_type != be16_t(Ethernet::Type::kMpls)) {
+      // Currently ignore non MPLS packets
+      continue;
```

Comment:
It seems a bit odd to me to have the output gate have a mix of processed MPLS packets and unprocessed other packets.  I can see the use case, but wonder if it might be better to have two output gates, one for processed MPLS packets and another for the rest.